pkgs/by-name/tg/tg-archive/package.nix

```
@@ -0,0 +1,47 @@
+{ lib
+, python3
+, fetchFromGitHub
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "tg-archive";
+  version = "1.1.3";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "knadh";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-mcobB/z+e4LzEbqELWlUzhbdV5RIM2iImeg9JdVQQZc=";
+  };
+
+  pythonRelaxDeps = true;
+
+  nativeBuildInputs = with python3.pkgs; [
+    pythonRelaxDepsHook
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    setuptools
+    telethon
+    jinja2
+    pyyaml
+    cryptg
+    pillow
+    feedgen
+    python-magic
+    pytz
+  ];
+
+  pythonImportsCheck = [
+    "tgarchive"
+  ];
+
+  meta = with lib; {
+    description = "A tool for exporting Telegram group chats into static websites like mailing list archives";
+    homepage = "https://github.com/knadh/tg-archive";
+    license = licenses.mit;
+    maintainers = with maintainers; [ pacien ];
+    mainProgram = "tg-archive";
```

```suggestion
  meta = {
    description = "A tool for exporting Telegram group chats into static websites like mailing list archives";
    homepage = "https://github.com/knadh/tg-archive";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ pacien ];
    mainProgram = "tg-archive";
```

using `with lib;` is discouraged #208242